sklearn/decomposition/tests/test_kernel_pca.py

```diff
@@ -285,6 +311,36 @@ def test_kernel_conditioning():
     assert np.all(kpca.lambdas_ == _check_psd_eigenvalues(kpca.lambdas_))
 
 
+@pytest.mark.parametrize("n_components", [4, 10, 20],
+                         ids="n_components={}".format)
```

Comment:
aren't the ids automatically generated anyway?

Reply:
they do not contain the name of the parameter that varies. So you'll read "4" not "n_components=4". I prefer this for my debug comfort when there are several parameters varying together, but here there is only one so that's really optional ;)